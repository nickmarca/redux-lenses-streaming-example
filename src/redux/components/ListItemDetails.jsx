import React from "react";
import PropTypes from "prop-types";

const ListItemDetailsRow = ({ label, value }) => (
  <li>
    <strong>{label}</strong>: {value}
  </li>
);

class ListItemDetails extends React.Component {
  constructor(props) {
    super(props);

    this.commitMessage = this.commitMessage.bind(this);
    this.clearSelectedItem = this.clearSelectedItem.bind(this);
  }

  commitMessage(message) {
    return () => {
      if (message) {
        this.props.onCommitMessage(message);
      }
    };
  }

  clearSelectedItem() {
    this.props.onShowRowDetails({});
  }

  render() {
    let arr = [];
    const { message } = this.props;
    const { value } = message;
    if (value) {
      Object.keys(JSON.parse(value)).forEach(function(k) {
        arr.push({ label: k, value: JSON.parse(value)[k] });
      });
    }
    return (
      <div>
        {arr.length > 0 ? (
          <div className="notification content list-item-details">
            <ul>
              {arr.map(item => (
                <ListItemDetailsRow
                  key={item.label}
                  label={item.label}
                  value={item.value ? String(item.value) : item.value}
                />
              ))}
            </ul>
            <button
              className="button is-info is-small"
              onClick={this.commitMessage(message)}
            >
              Commit
            </button>
            <button
              className="button is-white is-small"
              onClick={this.clearSelectedItem}
            >
              Hide details
            </button>
          </div>
        ) : (
          ""
        )}
      </div>
    );
  }
}

ListItemDetails.defaultProps = {
  message: {}
};

ListItemDetails.propTypes = {
  message: PropTypes.object,
  onShowRowDetails: PropTypes.func.isRequired
};

export default ListItemDetails;
