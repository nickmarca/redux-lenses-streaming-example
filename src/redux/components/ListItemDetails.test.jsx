import React from 'react';
import { create } from 'react-test-renderer';
import ListItemDetails from './ListItemDetails';

describe('ListItemDetails', () => {
    it('matches the snapshot', () => {
        const listItemDetails = create(<ListItemDetails />);

        expect(listItemDetails.toJSON()).toMatchSnapshot();
    });
});
