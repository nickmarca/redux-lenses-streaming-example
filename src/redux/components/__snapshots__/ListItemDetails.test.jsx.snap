// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemDetails matches the snapshot 1`] = `
<div>
  
</div>
`;
